include/RAJA/LegacyCompatibility.hxx

```diff
@@ -0,0 +1,215 @@
+#ifndef KRIPKE_LEGAGY_COMPATIBILITY_HXX
+#define KRIPKE_LEGAGY_COMPATIBILITY_HXX 1
+
+static_assert(__cplusplus >= 201103L, "C++ standards below 2011 are not supported");
+
+#include <type_traits>
+#include <utility>
+#include <iostream>
+#include <functional>
+#include <cstdint>
+
+namespace VarOps {
```

Comment:
On this note: John Mellor-Crummey was asking whether we could somehow tag the top-level entry points for RAJA so he can do better collapsing of nested/inlined calls in HPCToolkit.  Basically he wants to show users a profile that only includes, e.g. RAJA:foraall but not all the plumbing underneath. 

I think you could do this by putting all the app entry points in the RAJA namespace and moving internal calls out to other namespaces. Then tools would have nice labels and could pick what to show. Thoughts?